Moar release docs (#9153)
ipfs/kubo: docs/RELEASE_ISSUE_TEMPLATE.md

    - [ ] Verify that version string in [`version.go`](https://github.com/ipfs/go-ipfs/tree/master/version.go) has been updated.
    - [ ] Merge `release-vX.Y.Z` into the `release` branch.
    - [ ] Tag this merge commit (on the `release` branch) with `vX.Y.Z`.
    - [ ] Publish to `dist.ipfs.io` (mostly automated stuff, you just have to push two slocs changes for dist's CI to do magic)
      - [ ] Open a PR with `./dist.sh add-version kubo vX.Y.Z` update on [ipfs/distributions](https://github.com/ipfs/distributions).
      - [ ] Merge that PR (ONCE CI IS GREEN).
      - [ ] Wait for CI to be green on master.
    - [ ] **Important release to do**
      - [ ] to [dist.ipfs.io](https://dist.ipfs.io) *cross check CI*
      - [ ] to [github](https://github.com/ipfs/go-ipfs/releases) ***manual action to do***
        - [ ] Publish a new github release, target the tag made previously. Include changelog highlights, just lookup previous major releases for inspiration.
        - [ ] Run the [synchronisation workflow](https://github.com/ipfs/kubo/actions/workflows/sync-release-assets.yml) (it also runs once a day on a cron job)

>       - [ ] Send a message to notify the gateway team.

Reviewer: ```suggestion
      - [ ] Send a message to notify #bifrost
```

>       - [ ] Send a message to notify the GUI team (need to do a new IPFS-desktop release).

Reviewer: ```suggestion
      - [ ] Send a message to notify #ipfs-gui-dev (since they need to do a new IPFS-desktop release).
```
